src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts

```diff
@@ -77,6 +78,37 @@ const unitListingQuery =
       subjectCategories,
       reshapedUnits,
     );
+
+    // this is a temporary fix for legacy data to remove duplicates units within key stage
+    // we will remove once legacy data is removed
+    const reduceBySlugAndYearOrder = (arr: GroupedUnitsSchema) => {
```

Comment:
This is sooooo much easier if you do it before the grouping of units. It shuld be just a couple of lines of code.